packages/ts-exporter/src/test/java/org/uberfire/jsbridge/tsexporter/dependency/DependencyGraphTest.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.uberfire.jsbridge.tsexporter.dependency;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import javax.lang.model.element.Element;
+
+import com.google.testing.compile.CompilationRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.uberfire.jsbridge.tsexporter.util.TestingUtils;
+
+import static java.util.Arrays.stream;
+import static java.util.Collections.singleton;
+import static java.util.Comparator.comparing;
+import static java.util.stream.Collectors.toList;
+import static org.junit.Assert.assertEquals;
+import static org.uberfire.jsbridge.tsexporter.decorators.DecoratorStore.NO_DECORATORS;
+import static org.uberfire.jsbridge.tsexporter.util.TestingUtils.element;
+import static org.uberfire.jsbridge.tsexporter.util.TestingUtils.memberElement;
+import static org.uberfire.jsbridge.tsexporter.util.TestingUtils.type;
+
+public class DependencyGraphTest {
+
+    @Rule
+    public final CompilationRule compilationRule = new CompilationRule();
+
+    @Before
+    public void before() {
+        TestingUtils.init(compilationRule.getTypes(), compilationRule.getElements());
+    }
+
+    class X {
+
+        X field;
+    }
+
+    @Test
+    public void testInvalidElements() {
+        final DependencyGraph graph = new DependencyGraph(Stream.empty(), NO_DECORATORS);
+        assertEquals(null, graph.add(null));
+        assertEquals(0, graph.vertices().size());
+        assertEquals(null, graph.add(memberElement("field", type(X.class))));
```

Comment:
`field` is a good candidate for constant

Reply:
That's just a placeholder name.